test/cli/diff/kbcdir-ignore/in/bar/transformation/keboola.python-transformation-v2/python-transformation/blocks/002-block/kbcdir.json

```diff
@@ -0,0 +1,3 @@
+{
+  "isIgnored": true
+}
```

Comment:
@webrouse a když teda přidám jednoduchý e2e test na diff, tak začne failovat na tom že ty soubory v ignorovaným adresáři jsou unknown paths. To je potřeba ještě nějak víc dořešit než jen tou úpravou `ReadSubDirs` v orch./transf. mapperu?

![CleanShot 2022-04-27 at 13 18 07](https://user-images.githubusercontent.com/215660/165507015-9513c6c4-6c33-41bf-8e67-6a44f1c1fae3.jpg)

Reply:
Když je ta transformace v `initial-state` a v `in` je k bloku `002-block` ten `kbcdir.json` s `isIgnored=true`, tak vlastně by jeho folder měl být prázdný kromě toho jsonu. 🤔 Nicméně pořád to ještě failuje:

![CleanShot 2022-04-27 at 13 31 45](https://user-images.githubusercontent.com/215660/165509031-e827ce2a-a3ae-4d35-879f-a1423ddd40b2.jpg)

Reply:
K 

1. neprisiel som  zatial na ziadne jednoduche riesenie, ale na nejake snad pridem :)
2. to je teda asi iny problem? Skus si vypisat na STDOUT, ktore adresare sa ignorovali, ... ci sa to dostane na miesto v kode, na ktore ocakavas

Reply:
Já myslím že to ignorování funguje dobře. Dir s blokem se nedostane do state a právě proto si nejspíš začne stěžovat, že jsou tam nějaký soubory navíc (ten `kbcdir.json` a folder ve kterým je) který neočekává. A zřejmě je teda asi potřeba ještě té věci která vyhodnocuje `Unknown paths found` vysvětlit že je má taky ignorovat?

![CleanShot 2022-04-27 at 13 42 31](https://user-images.githubusercontent.com/215660/165510845-b04994dc-0ea2-4063-9422-83dd7b0b52df.jpg)
![CleanShot 2022-04-27 at 13 47 22](https://user-images.githubusercontent.com/215660/165511633-15c9ab2b-d083-4a28-a23c-bfc10fa77372.jpg)

Reply:
No jo, ale vlastně ten kód ignorovanýho bloku by neměl být ani vidět v `Unexpected STDOUT`. 🤔

Reply:
`nedostane do state a právě proto si nejspíš začne stěžovat,` + `vysvětlit že je má taky ignorovat?`  jj toto je jasne, ide o to, len ako to spravit jednoducho.

Suhlasim s tebou, ze to funguje podla vsetkeho spravne, v unexpected STDOUT mas `-`, ak pozeram dobre, to znamena, ze je to pritomne iba v `remote` state.

Reply:
Konkretne sa ignorovanie nejakej path z `knownpaths` riesi tu:
https://github.com/keboola/keboola-as-code/blob/d7cbd2f2aaa6b0388f591238c9b3d24bf44f3f86/internal/pkg/filesystem/knownpaths/paths.go#L239-L245

Reply:
Tam akurat rozmyslam nad tym, ako to spravit, aby to fungovalo aj v sablone, .... ci tam pridat ako parameter `FileLoader` alebo inak .... este musim nad tym chvilku posediet.

Reply:
@JakubMatejka Mohol by si prosim do `FileLoader`  interface/implementacie pridat novu metodu 
`IsIgnored(path string) (string, error)`?

A tam presunut danu cast logiky? 
Tj, ak `path` bude adresar, a bude mat `kbcdir.json`, v ktorom je `isIgnored=true`, tak tato metoda vrati `true` (ano, je to ignorovana path).

Ja by som zatial upravil `knownpaths` aby sa tam dal externe nastavit filter.
To bude asi najcistejsie riesenie, vramci aktualnych moznosti.

Reply:
ok

Reply:
V `state.go` sa to potom spoji:
https://github.com/keboola/keboola-as-code/blob/d7cbd2f2aaa6b0388f591238c9b3d24bf44f3f86/internal/pkg/state/state.go#L71-L94

Reply:
(este som ^^^ opravil preklep v return type na `bool`)

Reply:
Je tam: https://github.com/keboola/keboola-as-code/pull/648/commits/4582df20695e15160c688caba498adc17ca979f9

Reply:
Ja cakam este na testy: https://github.com/keboola/keboola-as-code/pull/651

Reply:
Unknown paths jsou fixnutý. 👍

Teď mi teda ještě pořád failuje e2e test. Chápu to teda dobře, že když je v `initial-state` transformace s bloky, jednomu z nich v `in` odstraním z adresáře všechny soubory a nechám tam jen ten `kbcdir.json` s `isIgnored=true`, tak by se teda neměl načíst do state a diff by ho měl ignorovat, ne? Takže tohle by se už nemělo zobrazit a mám tam bug?

![CleanShot 2022-04-27 at 16 00 06](https://user-images.githubusercontent.com/215660/165536269-305c1a85-bac1-4d70-84e3-fc0cf9c1862d.jpg)